woocommerce/woocommerce-ios
# Use Sentry instead of Fabric

**To Test:**

1) Check out this branch
2) Run `cd ~/.mobile-secrets && git pull && cd - && bundle exec fastlane run configure_apply` to update your local configuration secrets.
3) Download [the attached patch file](https://github.com/woocommerce/woocommerce-ios/files/3234545/0001-Add-crash-logging-tests.txt), and run `git apply $path_to_file`.
4) In Xcode, do a workspace search for `shouldSendEvent(_` and choose the first result. Comment out the debug-specific code so that it looks like this:
```swift
//        #if DEBUG
//        let result = false
//        #else
        let result = !CrashLogging.userHasOptedOut
//        #endif
```

5) Run the app on-device.
6) Disconnect the device from the host machine, and force-quit the app.
7) Relaunch the app and Log in, ensuring that your user is set up to send crash reports (by default, this will be disabled).
8) Open Settings. Note that a new test event shows up in Sentry almost immediately  – you can see it under "Events" if you filter by woocommerce-ios. It should have your username associated with it.
9) Navigate to Settings > Feature Request (the app will crash)
10) Re-run the app, and let it idle for a moment (this'll ensure the crash report goes to Sentry)
11) Note your shiny new crash event in Sentry (it may take ~60 seconds to show up) – you can see it under "Events" if you filter by woocommerce-ios. It should have your username associated with it, as well as a full stack trace, and the device locale.

Update release notes:
